aws-iot-securityprofile/aws-iot-securityprofile.json

```diff
@@ -0,0 +1,359 @@
+      "type": "string",
+      "pattern": "[a-zA-Z0-9:_-]+",
+      "minLength": 1,
+      "maxLength": 128
+    },
+    "SecurityProfileDescription": {
+      "description": "A description of the security profile.",
+      "type": "string",
+      "maxLength": 1000
+    },
+    "Behaviors": {
+      "description": "Specifies the behaviors that, when violated by a device (thing), cause an alert.",
+      "type": "array",
+      "maxLength": 100,
+      "uniqueItems": true,
+      "insertionOrder": false,
+      "items": {
+        "$ref": "#/definitions/Behavior"
+      }
+    },
+    "AlertTargets": {
+      "description": "Specifies the destinations to which alerts are sent.",
+      "type": "object",
+      "patternProperties": {
+        "[a-zA-Z0-9:_-]+": {
+          "$ref": "#/definitions/AlertTarget"
+        }
+      },
+      "additionalProperties": false
+    },
+    "AdditionalMetricsToRetain": {
+      "description": "This parameter has been deprecated, please use AdditionalMetricsToRetainV2.",
+      "type": "array",
+      "uniqueItems": true,
+      "insertionOrder": false,
+      "items": {
+        "type": "string",
+        "pattern": "[a-zA-Z0-9:_-]+",
+        "minLength": 1,
+        "maxLength": 128
+      }
+    },
+    "AdditionalMetricsToRetainV2": {
+      "description": "A list of metrics whose data is retained (stored). By default, data is retained for any metric used in the profile's behaviors, but it is also retained for any metric specified here.",
+      "type": "array",
+      "uniqueItems": true,
+      "insertionOrder": false,
+      "items": {
+        "$ref": "#/definitions/MetricToRetain"
+      }
+    },
+    "Tags": {
+      "description": "Metadata that can be used to manage the security profile.",
+      "type": "array",
+      "maxItems": 50,
+      "uniqueItems": true,
+      "insertionOrder": false,
+      "items": {
+        "$ref": "#/definitions/Tag"
+      }
+    },
+    "TargetArns": {
+      "description": "A set of target ARNs that the security profile is attached to.",
+      "type": "array",
+      "uniqueItems": true,
+      "insertionOrder": false,
+      "items": {
+        "description": "The ARN of the target to which the security profile is attached.",
+        "type": "string",
+        "maxLength": 2048
+      }
+    },
+    "SecurityProfileArn": {
+      "description": "The ARN (Amazon resource name) of the created security profile.",
+      "type": "string"
+    }
+  },
+  "additionalProperties": false,
+  "primaryIdentifier": [
+    "/properties/SecurityProfileName"
+  ],
+  "required": [
+    "SecurityProfileName"
+  ],
+  "createOnlyProperties": [
+    "/properties/SecurityProfileName"
+  ],
+  "readOnlyProperties": [
+    "/properties/SecurityProfileArn"
+  ],
+  "handlers": {
+    "create": {
+      "permissions": [
+        "iot:CreateSecurityProfile",
+        "iot:AttachSecurityProfile",
+        "iam:PassRole"
```

Comment:
Why do we need the permission for iam:PassRole for create and update calls?

Reply:
This is needed if the customer specifies an AlertTarget. Device Defender (as all other AWS services) requires customers to have a permission for passing the role.